contracts/non-fungible-tokens/andromeda-crowdfund/src/contract.rs

```diff
@@ -288,6 +300,151 @@ fn execute_start_campaign(
     Ok(resp)
 }
 
+fn execute_purchase_tiers(
+    ctx: ExecuteContext,
+    orders: Vec<SimpleTierOrder>,
+) -> Result<Response, ContractError> {
+    let ExecuteContext {
+        ref deps, ref info, ..
+    } = ctx;
+
+    // Ensure campaign accepting coin is received
+    let payment: &Coin = &info.funds[0];
+    let campaign_config = get_config(deps.storage)?;
+
+    ensure!(
```

Comment:
It would be better to have that check before `let payment: &Coin = &info.funds[0];`